dexlib2/src/main/java/com/android/tools/smali/dexlib2/writer/DexWriter.java

```diff
@@ -1256,6 +1256,8 @@ private int writeCodeItem(@Nonnull DexDataWriter writer,
                     if (paramCount > outParamCount) {
                         outParamCount = paramCount;
                     }
+                } else if (instruction.getOpcode().referenceType == ReferenceType.CALL_SITE) {
+                    outParamCount = ((VariableRegisterInstruction)instruction).getRegisterCount();
```

Comment:
Judging from the code above, I think this should be
```
                    int paramCount = ((VariableRegisterInstruction)instruction).getRegisterCount();
                    if (paramCount > outParamCount) {
                        outParamCount = paramCount;
                    }
```
I suggest moving the declaration of `paramCount` one level out and share the update to `outParamCount`. A test should be able to reveal how this should be.